pkg/epp/plugins/typedname.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugins
+
+// TypedName is a utility struct providing a type and a name
+// to plugins.
+// It implements the Plugin interface and can be embedded in
+// plugins across the code to reduce boilerplate.
+type TypedName struct {
+	// Type returns the type of the plugin.
+	Type string
+	// Name returns the name of this plugin instance.
+	Name string
+}
+
+const (
+	Separator = "/"
+)
+
+// String returns the type and name rendered as
+// "<type>/<name>".
+func (tn *TypedName) String() string {
+	return tn.Type + Separator + tn.Name
+}
+
+func (tn *TypedName) GetTypedName() TypedName {
```

Comment:
I understand the motivation behind this, but this function looks not natural.

Reply:
I’d argue it’s better to put back a one liner TypedName() function in each plugin.. 
yes, it adds “boilerplate” code (hard to say this on one line), but more aligned with how things are usually written in go.

Reply:
I think your point is that a Plugin `HAS-A` TypedName and not `IS-A` TypedName (which is what embedding models).
We could add a `BasePlugin` struct that implements GetTypedName() generically and then embed that into Plugin, but I would prefer to delay that until there's more "behavior" in the base plugin.